### PR TITLE
feat: Add support for new select menus components

### PR DESCRIPTION
**Pull request by Rapougnac**
# Description
Add support for new types of select menus

## Connected issues & other potential problems
- https://github.com/nyxx-discord/nyxx_interactions/pull/62

## Type of change

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] Ran `dart analyze` or `make analyze` and fixed all issues
- [x] Ran `dart format --set-exit-if-changed -l 160 ./lib` or `make format` and fixed all issues
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my changes haven't lowered code coverage

### CODE REVIEW

File: lib/src/core/message/components/message_component.dart [hunk 1]
```diff
@@ -77,16 +94,22 @@ class MessageComponentEmoji implements IMessageComponentEmoji {
...
+  IMessageComponent._(RawApiMap raw) {
+    customId = raw['custom_id'] as String;
+  }
 }
 
 /// Generic container for components that can be attached to message
-abstract class MessageComponent implements IMessageComponent {
-  /// [ComponentType]
+abstract class MessageComponent extends IMessageComponent {
+  /// Type of component.
```
Comment (abitofevrything) [inline lib/src/core/message/components/message_component.dart:108]: Any implementation should go in the non-interface class (without an `I` prefix). Therefore all implementation classes should implement and not extend their interfaces, and any functionality should go in that class.

**Review by abitofevrything — CHANGES_REQUESTED** [review]

File: lib/src/core/message/components/message_component.dart (hunk 1 above)
Comment (Rapougnac) [reply lib/src/core/message/components/message_component.dart:108]: Oops, my bad, you're right, seems I've messed up my head writing too much interfaces..

**Review by abitofevrything — DISMISSED** [review]

File: lib/src/core/message/components/message_component.dart [hunk 2]
```diff
@@ -98,17 +126,43 @@ abstract class MessageComponent implements IMessageComponent {
...
+  late final int maxValues;
+
+  /// Whether this multiselect is disabled.
+  late final bool isDisabled;
+
+  MultiSelectAbstract(RawApiMap raw) : super(raw) {
+    placeholder = raw['placeholder'] as String?;
+    minValues = raw['min_values'] as int? ?? 1;
+    maxValues = raw['max_values'] as int? ?? 1;
+    isDisabled = raw['disabled'] as bool? ?? false;
```
Comment (l7ssha) [inline lib/src/core/message/components/message_component.dart:160]: are these defaults based on something or are arbitrary?

**Review by l7ssha — APPROVED** [review]

File: lib/src/core/message/components/message_component.dart (hunk 2 above)
Comment (Rapougnac) [reply lib/src/core/message/components/message_component.dart:160]: They're based on Discord's behaviour, whereas by default, a select menu can take only 1 to 1 (min, max) values, and are enabled.

Comment (l7ssha) [reply lib/src/core/message/components/message_component.dart:160]: good

**Pull request merged**